fail2ban/server/banmanager.py

```diff
@@ -319,22 +311,39 @@ def _inBanList(self, ticket):
 	# @return the list of ticket to unban
 	
 	def unBanList(self, time):
-		try:
-			self.__lock.acquire()
+		with self.__lock:
 			# Permanent banning
 			if self.__banTime < 0:
 				return list()
 
-			# Gets the list of ticket to remove.
-			unBanList = [ticket for ticket in self.__banList if ticket.isTimedOut(time, self.__banTime)]
-			
+			# Check next unban time:
+			if self.__nextUnbanTime > time:
+				return list()
+
+			# Gets the list of ticket to remove (thereby correct next unban time).
+			unBanList = {}
+			self.__nextUnbanTime = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFL
```

Comment:
probably doesn't matter in the "long run" but why bother with a long here -- wouldn't largest int be sufficient?  comparisons seems to be 2 times faster:

```ipython
*In [20]: long_ = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFL

In [21]: int_ = 0xFFFFFFFFFFFFFFF

In [22]: timeit long_ > 1
10000000 loops, best of 3: 65.2 ns per loop

*In [23]: timeit int_ > 1
10000000 loops, best of 3: 28.5 ns per loop
```